### DISCUSSION TITLE
Envoy cannot make requests to authorization-service

### DISCUSSION THREAD

**Discussion by edocrippaofficial** [mia-platform/community#54] [Q&A] [answered]
I replaced the API-Gateway based on nignx with the one that uses Envoy, but I wasn't able to handle the requests (with every request, the reply is a 404 error).
After some log searching, I found that the external authorization filter of Envoy is trying to contact the authorization-service, but the path is wrong.

Here is an example: 
I made a request to `http://my-site/v2/payments/`
As the documentation says, the `authorization-service` exposes the path `/auth`, and it expects the path and the HTTP verb as headers, but here i see that Envoy is trying to call `/auth/v2/payments/`, a path that does not exists.

Is this the desired behavior? 
How can I make the request to the path `/auth`?
How does the interaction between Envoy and the authorization-service works?

Here are some log from the service:
```
[2022-10-14T12:38:03.691502Z] DEBUG:
    scope: "router"
    message: "[C0][S4143300971808305331] cluster 'authorization-service' match for URL '/auth/v2/payments/'"
```
```
[2022-10-14T12:38:03.693811Z] DEBUG:
    scope: "http"
    message: "async http request response headers (end_stream=false):\n':status', '404'\n'content-type', 'text/plain; charset=utf-8'\n'x-content-type-options', 'nosniff'\n'date', 'Fri, 14 Oct 2022 12:38:03 GMT'\n'content-length', '19'\n'x-envoy-upstream-service-time', '1'\n"
```
```
[2022-10-14T12:38:03.694917Z] DEBUG:
    scope: "http"
    message: "[C11][S1001450323973949704] Sending local reply with details ext_authz_denied"
```

**Comment by davidebianchi** [comment, marked as answer]
Hi @EdoCrippaOfficial! Yes, this is the intended behaviour. 
What version of the authorization-service do you have? Because at least the [v2.4.0](https://docs.mia-platform.eu/docs/runtime_suite/authorization-service/changelog#240---12-04-2022) is needed to work with envoy.

**Reply by edocrippaofficial** [reply]
2.3.0
I searched in the envoy and auth-service docs but I didn't find this information, nor the new calling style
Now that you point it out I found it in the auth-service changelog 
Maybe we can make it more highlighted in the doc, since it won't work otherwise
Thanks for now :)
